test/spec/http/request.ts

```diff
@@ -411,99 +394,158 @@ describe('HTTP Requestor', () => {
+        expect(httpRequestClient).toHaveBeenCalledWith(undefined, url, {
+          data: undefined,
+          headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+            'X-Okta-User-Agent-Extended': USER_AGENT
+          },
+          withCredentials: false
+        });
+        const res = await requestPromise;
+        expect(res).toBe(response1);
       });
-      await advanceTestTimers();  
-      expect(httpRequestClient).toHaveBeenCalledTimes(2);
-      expect(httpRequestClient).toHaveBeenCalledWith(undefined, url, {
-        data: undefined,
-        headers: {
-          'Accept': 'application/json',
-          'Content-Type': 'application/json',
-          'X-Okta-User-Agent-Extended': USER_AGENT
-        },
-        withCredentials: false
+  
+      it('should retry on network error 3 times maximum', async () => {
+        httpRequestClient = jest.fn()
+          .mockRejectedValueOnce(new TypeError('Load failed'))
+          .mockRejectedValueOnce(new TypeError('Load failed'))
+          .mockRejectedValueOnce(new TypeError('Load failed'))
+          .mockRejectedValueOnce(new TypeError('Load failed'))
+          .mockResolvedValueOnce({
+            responseText: JSON.stringify(response1)
+          });
+        createAuthClient({pollDelay: 500});
+        expect(document.hidden).toBe(false);
+        let didThrow = false;
+        const requestPromise = httpRequest(sdk, {
+          url,
+          pollingIntent: true,
+        }).catch(error => {
+          didThrow = true;
+          expect((error as Error).message).toBe('Load failed');
+        });
+        await advanceTestTimers();
+        expect(httpRequestClient).toHaveBeenCalledTimes(4);
+        expect(httpRequestClient).toHaveBeenCalledWith(undefined, url, {
+          data: undefined,
+          headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+            'X-Okta-User-Agent-Extended': USER_AGENT
+          },
+          withCredentials: false
+        });
+        await requestPromise;
+        expect(didThrow).toBe(true);
       });
-      const res = await requestPromise;
-      expect(res).toBe(response1);
     });
 
-    it('should retry on network error 3 times maximum', async () => {
-      httpRequestClient = jest.fn()
-        .mockRejectedValueOnce(new TypeError('Load failed'))
-        .mockRejectedValueOnce(new TypeError('Load failed'))
-        .mockRejectedValueOnce(new TypeError('Load failed'))
-        .mockRejectedValueOnce(new TypeError('Load failed'))
-        .mockResolvedValueOnce({
-          responseText: JSON.stringify(response1)
+    describe('pollDelay = 0 (default)', () => {
+      it('should NOT wait for document to be visible before making request', async () => {
+        createAuthClient();
+        expect(document.hidden).toBe(false);
+  
+        togglePageVisibility();
```

Comment:
I think this test shouldn't toggle page visibility. It should confirm the flow works without it